modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java

```diff
@@ -199,4 +205,16 @@ public String getHttpHostAddress() {
     public InetSocketAddress getTcpHost() {
         return new InetSocketAddress(getHost(), getMappedPort(ELASTICSEARCH_DEFAULT_TCP_PORT));
     }
+
+    /**
+     * Configure max heap size, using ES_JAVA_OPTS environment variable
+     *
+     * @param maxHeapSizeInBytes Maximum and start heap size in bytes
+     * @return this
+     */
+    public ElasticsearchContainer withMaxHeapSizeInBytes(long maxHeapSizeInBytes) {
+        String options = String.format("-Xms%d -Xmx%d", maxHeapSizeInBytes, maxHeapSizeInBytes);
+        withEnv("ES_JAVA_OPTS", options);
+        return this;
+    }
```

Comment:
This would overwrite `ES_JAVA_OPTS` if it is already set.
Also, if somebody calls `withMaxHeapSizeInBytes(123).withEnv("ES_JAVA_OPTS", "...")`, the value won't be set either.

We should instead store the value to a variable and then configure the env variable in `configure()` (note that there is `withEnv(String, Function<Optional<String>, String>)` method that allows appending to an existing value if already set)